operators/pkg/exmagnt/instance.go

```diff
@@ -0,0 +1,270 @@
+// Copyright 2020-2022 Politecnico di Torino
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package exmagnt contains the main logic and helpers
+// for the crownlabs exam agent component.
+package exmagnt
+
+import (
+	"bytes"
+	_ "embed"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"path"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2"
+	"k8s.io/utils/pointer"
+	ctrl "sigs.k8s.io/controller-runtime"
+	ctrlutil "sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	clv1alpha2 "github.com/netgroup-polito/CrownLabs/operators/api/v1alpha2"
+	"github.com/netgroup-polito/CrownLabs/operators/pkg/forge"
+)
+
+//go:embed redirecting.html
+var httpPageStartingUp string
+
+// EAInstance represents an Instance within the exmagnt.
+type EAInstance struct {
+	ID                string                               `json:"id"`
+	Template          string                               `json:"template"`
+	Running           *bool                                `json:"running,omitempty"`
+	CustomizationUrls clv1alpha2.InstanceCustomizationUrls `json:"customizationUrls"`
+}
+
+// EAInstanceHandler is the handler for the EAInstance.
+type EAInstanceHandler struct{}
```

Comment:
Would save here the client, instead of using the singleton. It simplifies future extensions.